[DOCS] A draft of documentation for Points of Interest (#5122)
ClanGenOfficial/clangen: docs/dev/writing/points-of-interest.md

The Points of Interest system adds more depth to each Clan's territory by giving them a randomized gathering place, moonplace, and three discoverable locations in their territory that serve as points of interest to be mentioned in patrols and short events. This project helps make Clans feel even more unique to each other.

Disclaimer: Points of Interest do <b>not</b> use exclusionary tags.

# Creating a Point of Interest

New points of interest can be created in resources/dicts/points_of_interest.json. They use the following format:

~~~     
    "ID": {
        "category": "",
        "biome": [""],
        "tags": []
    }
~~~

> The ID should be formatted as what category of Point of Interest it is and something to describe  the location itself. 

Reviewer: ```suggestion
The ID should be formatted as the category it is and a term to describe the location itself. For example: `gathering_meteor` is a gathering place most defined by a meteor crater.
```

Biomes are listed the same way they are for short events and patrols, by using the biome's name in quotation marks or "any".

> Options for categories and tags are listed in a section below, each should be within quotation marks with a comma between them (if necessary).

Reviewer: ```suggestion
Options for categories and tags are listed in [categories and tags](#categories-and-tags) below, each should be within quotation marks with a comma between them (if necessary).
```

Here is an example of how a Point of Interest is coded.

~~~
    "moon_meteor": {
        "category": "moonplace",
        "biome": ["any"],
        "tags": []
    }
~~~

> After adding a Point of Interest to their dicts file, it must also be added to resources/lang/en/points_of_interest.en.json. This allows you to determine what the Point of Interest is displayed as in game.

Reviewer: ```suggestion
After adding a Point of Interest to their dicts file, it must also be added to `resources/lang/en/points_of_interest.en.json`. This allows you to determine what the Point of Interest is displayed as in game.
```

Points of Interest are distributed into three categories. Each Territory has randomly generated Points of Interest in each category.

|  Category |                                                                                            Description                                                                                           |
|:---------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
| gathering | This is the Clan's Gathering place, where they interface with other Clans. One exists per territory.                                                                                             |
| moonplace | This is the Clan's moon place, where medicine cats meet once a moon. One exists per territory. It typically features some connection to the moon, stars, or night sky. One exists per territory. |
| territory | This is a feature within the Clan's territory that is notable due to either its appearance or opportunities around it. Three can be discovered per territory.                                    |

## Tags 

> Points of Interests can fall into many different tags that denote circumstances around them to help specify how it can be used and what events/patrols it would fit with.

Reviewer: ```suggestion
Points of Interests can use many different tags that denote circumstances around them to help specify how it can be used and what events/patrols it would fit with.
```

|       prey        |                     anything the Clan hunts. Has multiple more specific tags. Should not be used together.                      |
|    prey:flying    |                                                 any prey that primarily flies.                                                  |
|    prey:water     |                                  prey found in or around water. assume your cats will get wet.                                  |
|    prey:ground    |                                                      ground-dwelling prey.                                                      |
|       rocks       |                                   feature that primarily includes rocks, boulders, or stone.                                    |
|      tainted      |                                  carries a risk of injury or illness due to unsafe conditions                                   |
|       trees       |                                                       incorporates trees.                                                       |

> |      Twolegs      |                          revolves around Twolegs. Has more specific tags. Should not be used together.                          |

Reviewer: ```suggestion
|      Twolegs      |                          revolves around Twolegs. Has more specific tags. Should not be used alongside other twoleg tags.                          |
```

| Twolegs:abandoned |                    An object, structure, area created or modified by Twolegs that has since been abandoned.                     |
|  Twolegs:present  |                   An object, structure, area created or modified by Twolegs that has regular Twoleg activity.                   |
|     unstable      | If there is a potential for the structure to collapse in some way, unstable ceilings, unstable ground, an old tree branch, etc. |
|       water       |                                         Lakes, streams, waterfalls, oceans, bays, etc.                                          |

# Using Points of Interest

## Constraints

Patrols and Short Events now have an additional constraint that can be utilized to include either a specific Point of Interest ID or tag. 

> You can add this to any short event or patrol to constrain by Point of Interest, however, remember to constrain either via names, or by a single tag.

Reviewer: ```suggestion
You can add this to any short event or patrol to constrain by Point of Interest. However, remember to constrain either via names or by a single tag, but not by both at once.
```

~~~
"poi": {
    "name": ["name"]
    "tags": ["tag"]
    }
~~~

## Using the Point Of Interest in a Sentence

> Using a system similar to pronoun tags, Points of Interests can be mentioned in Short Events or Patrols with {POI} followed by relevant information, either the Points of Interest's IDs, or tags for a pool of Points of Interest.

Reviewer: ```suggestion
Using a system similar to pronoun tags, Points of Interests can be mentioned in Short Events or Patrols with {POI} followed by relevant information; either the Points of Interest's IDs, or tags for a pool of Points of Interest.
```
